File: ChapterThree/selection_sort.py
def selection_sort(arr, n):
    for i in range(n - 1):
        smallest = i
        for j in range(i + 1, n):
            if arr[j] < arr[smallest]:
                smallest = j
        tmp = arr[i]
        arr[i] = arr[smallest]
        arr[smallest] = tmp


arr = [3, 2, 4, 1, 8, 6, 9, 7]
selection_sort(arr, len(arr))
print(arr)

File: ChapterThree/selection_sort.rs
fn selectionSort(arr:[i32]){
    for i in 0..arr.len()-1 {
        let smallest: usize = i;
        for j in i+1..arr.len() {
            if arr[j] < arr[smallest] {
                smallest = j;
            }
        }
        let tmp: i32 = arr[i];
        arr[i] = arr[smallest];
        arr[smallest] = tmp
    }
}

fn main(){
    let xs = [3, 2, 4, 1, 8, 6, 9, 7];
    selectionSort(xs);
    print!("[");
    for elem in xs {
        print!("{},",elem);
    }
    print!("]");
}